include/Listeners.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+namespace Listeners {
+
+  #define SET_LISTENER(TYPE, SUBTYPE, NAME, TARGET)			\
+    NAME.notify = [](wl_listener *listener, void *data)			\
+  		{							\
+  		  TYPE *that(static_cast<TYPE *>(wl_container_of(listener, static_cast<SUBTYPE *>(nullptr), NAME))); \
+  		  that->TARGET(listener, data);				\
+  		};							\
+
+    struct ServerCursorListeners
```

Comment:
Why put all the struts outside of the corresponding classes' files ?

Reply:
It made more sens to me to isolate the SET_LISTENER macro such as all the listeners that came with it in one dedicated file where you can find again all the listeners you need

Reply:
I feel it makes it harder to know what's inside a class. Also each time you include this file you pull a lot of useless dependencies.

Reply:
(speaking of structs ofc.)

Reply:
You're make a good point here..
So what about the SET_LISTENER macro ?

Reply:
It should stay here.